keras/layers/containers.py

```diff
@@ -251,22 +272,35 @@ def reset_states(self):
             if hasattr(l, 'reset_states') and getattr(l, 'stateful', False):
                 l.reset_states()
 
-    def set_previous(self, layer, connection_map={}):
+    def set_previous(self, layer, connection_map={}, init_weights=True):
```

Comment:
There is already a variable named `initial_weights`. To avoid confusion, please use `reset_weights`.